shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/core/util/PipelineJobPreparerUtils.java

```diff
@@ -0,0 +1,170 @@
+                                                           final PipelineDataSourceManager dataSourceManager) throws SQLException {
+        if (null != initProgress) {
+            Optional<IngestPosition<?>> position = initProgress.getIncremental().getIncrementalPosition(dumperConfig.getDataSourceName());
+            if (position.isPresent()) {
+                return position.get();
+            }
+        }
+        String databaseType = dumperConfig.getDataSourceConfig().getDatabaseType().getType();
+        DataSource dataSource = dataSourceManager.getDataSource(dumperConfig.getDataSourceConfig());
+        return PositionInitializerFactory.getInstance(databaseType).init(dataSource);
+    }
+    
+    /**
+     * Check data source.
+     *
+     * @param databaseType database type
+     * @param dataSources data source
+     */
+    public static void checkSourceDataSource(final String databaseType, final Collection<? extends DataSource> dataSources) {
+        if (null == dataSources || dataSources.isEmpty()) {
+            log.info("source data source is empty, skip check");
+            return;
+        }
+        DataSourceChecker dataSourceChecker = DataSourceCheckerFactory.getInstance(databaseType);
+        dataSourceChecker.checkConnection(dataSources);
+        dataSourceChecker.checkPrivilege(dataSources);
+        dataSourceChecker.checkVariable(dataSources);
+    }
+    
+    /**
+     * Check target data source.
+     *
+     * @param databaseType database type
+     * @param importerConfig importer config
+     * @param targetDataSources target data sources
+     */
+    public static void checkTargetDataSource(final String databaseType, final ImporterConfiguration importerConfig, final Collection<? extends DataSource> targetDataSources) {
+        DataSourceChecker dataSourceChecker = DataSourceCheckerFactory.getInstance(databaseType);
+        if (null == targetDataSources || targetDataSources.isEmpty()) {
+            log.info("target data source is empty, skip check");
+            return;
+        }
+        dataSourceChecker.checkConnection(targetDataSources);
+        dataSourceChecker.checkTargetTable(targetDataSources, importerConfig.getTableNameSchemaNameMapping(), importerConfig.getLogicTableNames());
+    }
+    
+    
+    /**
+     * Cleanup job preparer.
+     *
+     * @param databaseType database type
+     * @param dataSourceType data source type
+     * @param dataSourceParameter data source parameter
+     * @throws SQLException sql exception
+     */
+    public static void cleanup(final DatabaseType databaseType, final String dataSourceType, final String dataSourceParameter) throws SQLException {
+        PositionInitializer positionInitializer = PositionInitializerFactory.getInstance(databaseType.getType());
+        PipelineDataSourceConfiguration pipelineDataSourceConfig = PipelineDataSourceConfigurationFactory.newInstance(dataSourceType, dataSourceParameter);
```

Comment:
1, Could we rename the `cleanup` method name? It's not clear enough.

2, Could we just transfer `PipelineDataSourceConfiguration pipelineDataSourceConfig` to replace `final DatabaseType databaseType, final String dataSourceType, final String dataSourceParameter`?